### DISCUSSION TITLE
ID & License System - How should it be?

### DISCUSSION THREAD

**Discussion by Tomboru** [EveronLife/EveronLife#97] [Polls]
The system would work like this ->
You have these things as an item, you can hold them in the hand and show it to others.
The other players can see all the information on the object that you are holding. (As an example ID)
The information can be changed for each player so every ID shows their name, etc.

Why not virtual? 
What if you want to leave the ID somewhere so police can't just get your name 
by pressing one button.

But we can still say every processer just knows it, even if you don't have a licence with you.
Also, you can get a new ID just at a city hall or something. (Or you just spawn again with it)

First Idea:
Make each licence its own item.
This is more realistic, but also very annoying because then you have to search for it and also it will take
a lot of space in the inventory.

2nd Idea:
Make one item, the "ID"
This will show you the name, birthday, sex, (maybe a picture?)
and all the licences. 

This would be easier to handle because it's just one item.

![image](https://user-images.githubusercontent.com/79318124/172073520-c636bcd0-3332-4331-8799-331c9364a5db.png)

This image is just an example of how it could be.
(No final UI / Item)

**Comment by roobr** [comment]
I definitely support not having a "virtual" license for the reason of RP not having your id on you and then potentially having to be fingerprinted if you have a criminal record?

Even if it was virtual still have the option to not be carrying it

**Comment by jasper-one** [comment]
I'm for Idea 2 if it's not that much harder to implement. As long as you spawn with it each time and it's persistent. Second, to that, I would go virtual.

**Comment by TheOriginalCrazyDuck** [comment]
I feel idea 2 would be better. In terms of design, if there was some way to put the licences on the back and give the player to see both sides, I feel that would be much neater

**Comment by Tomboru** [comment]
Important to say.
I talked with Steve and he gave me some thoughts from his view.
If a server has like 20 licenses it will be hard to make it all on one card.
Or everything has to be very small.

Right now I'm working on a proper design that would fit well in the framework, but also most servers.
Else we have to say a player can only have like 10 licenses after that he will replace the first, second, etc.

**Reply by Arkensor** [reply]
Yeah, this is a problem that came to my mind while reading the pitch. People will likely have lots of licenses. I think we should **not** include any license information on the ID card. I think RP-wise the processors or vendors could have lists of who is licensed to trade with them. I am not really sure if you need to proof that you have a license for something to other players. But it would of course be a neat option. I think we could have multiple license cards as inventory items, and offer a small bag / purse maybe where you can put money and licenses (including drivers license) in and when asked you can pull out what you need from that. If you lose the license card you could get a new one issues for money. And people could find your license card but it is not issued for their name - but they could still try to play imposter with it.

I think a purse would be the best option. 20 cards in your inventory is whack, but you would rarely need to see or interact with them, and until then you got them all combined into this one 1x1 container in your inventory.

**Reply by Tomboru** [reply]
Yes so, it will be one item.
Yesterday an idea came up that you can switch the page with "r" and then it shows the next page of licences.
The Licences are just more for police, so when you are doing... Gold? And police stops you, you have to show the licence that you are allowed to.

The processer can just know it or if its possible we could save the metadata on the idcard and you need the idcard with you to process.
This way you can also like give your licence to other players, etc.

**Reply by thebonbon** [reply]
If you hold the book in your hand we could add an action for other players to "read" the book and that shows a UI. In the UI you could have different pages that you can change. Like one for farming licences, one for ID, Drivers licence etc..

**Reply by 12gabriel3** [reply]
I think not all licenses should be displayed, only the ones police should verify such as gun license.

**Comment by Tomboru** [comment]
![image](https://user-images.githubusercontent.com/79318124/172944914-37473194-6b00-4b4e-82e8-22ec581aa9c5.png)

This is right now how far I am with the ID Card

**Comment by borknbeans** [comment]
Based off of Tomboru's new implementation of Idea 2 where there is a book and the player can hold up the item allowing people to verify, should we

**A.** Allow players to drop their ID, allowing for anonymity when commiting crimes. They can later buy a new ID for a fee. This also opens up the option of players carrying other players ID's and pretending to be them
**B.** Players are required to have an ID on them and we put checks to prevent them from dropping, etc.

I think option A opens up a lot of cool RP opportunities, but it would require more things to be implemented to allow for a player to be verified. For example, in a scenario where the police capture a player commiting a crime, they need to charge that person and what if the person does not have an ID and refuses to tell them who they are. To fix this scenario I believe we could implement some sort of way for the police to bring the player to Jail/Police station, etc and "scan" their fingerprints which will bring up their information regardless of whether they have an ID. This could also bring up any warrants, past charges, etc.. 

If we have a system like above this could also open up opportunities for police to have to take players back to the station to verify their identity if they want to charge them for a crime and potentially charge them for having a "fake id" or "impersonation". Im curious of the communities thoughts on this.

Additionally, to prevent players from never carrying around an ID we can require it for accessing certain things i.e. to access your virtual garage you need YOUR ID, to purchase a weapon you need YOUR ID, etc.. This gives players reasons to continue to carry it around.

**Reply by Tomboru** [reply]
I would love to see option A.

Because you can also do more stuff like: 
Making a legal character that got in the country legally.

But also making a criminal character that is illegally in the country.

What would that do? Someone that is "legally" in the country will be saved in the police database.
Someone that is "illegally" in the country is not saved in the database => 
When police will notice that you will get a big fine or something and will be automatically added to the database.


The only thing that I hope will be possible, is that every player gets a unique face.
So then we could like work with a picture on the ID. 

This would be more work for me, but I would love to try it.

**Reply by Arkensor** [reply]
Option A yes. I think this will give us lots of possibilities for cops vs robbers' interaction later. Your ID is a physical item you often need. If you lose it you go back to the local town hall to get issued a new one (for a small amount of money). Police can tell if someone is the same person on the ID or not. Name and maybe even picture? We could maybe get into forged IDs as a gameplay element later as well.

**Reply by jasper-one** [reply]
Option A for sure. It would be amazing if there was a way to input something like a photo, maybe an **item preview** could even be used to show a different model like the player model?
